src/npc.cpp

```diff
@@ -1700,6 +1701,13 @@ void npc::shop_restock()
         }
     }
 
+    // This removes some items according to item spawn scaling factor,
+    const float spawn_rate = get_option<float>( "ITEM_SPAWNRATE" );
+    ret.remove_if( [spawn_rate]( auto & i ) {
+        ( void )i; // silence unused variable warning
```

Comment:
If you have to have `i`, this is preferred - but do you have to have i (I don't know)?
```suggestion
    ret.remove_if( [spawn_rate]( auto &/*i*/ ) {
```

Reply:
I tried doing it without any parameters `ret.remove_if( [spawn_rate]( ) {` and compiler did not accept that, so I assumed it needs i. It also accepted your corrections.